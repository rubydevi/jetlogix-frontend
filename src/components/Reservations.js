import React, { useContext, useEffect } from 'react';
import { useDispatch, useSelector } from 'react-redux';
import AuthContext from '../context/AuthProvider';
import { fetchReservedAeroplanes } from '../redux/aeroplanes/aeroplanesActions';
import AeroplaneItem from './AeroplaneItem';

const Reservations = () => {
  const dispatch = useDispatch();
  const userAeroplanes = useSelector((state) => state.aeroplanes.userReservedAeroplanes);
  const loading = useSelector((state) => state.aeroplanes.reservedAeroplanesLoading);
  const error = useSelector((state) => state.aeroplanes.reservedAeroplanesError);

  const userId = useContext(AuthContext).id;
  console.log(userId);

  useEffect(() => {
    dispatch(fetchReservedAeroplanes(userId));
  }, [dispatch, userId]);

  if (loading) {
    return (
      <div className="d-flex justify-content-center align-items-center vh-100">
        <div className="spinner-border" role="status">
          <span className="visually-hidden">Loading...</span>
        </div>
      </div>
    );
  }

  if (error) {
    return (
      <div className="d-flex justify-content-center align-items-center vh-100">
        Error:
        {error}
      </div>
    );
  }

  return (
    <div className="d-flex justify-content-center align-items-center vh-100">
      <div>
        <h1 className="d-flex justify-content-center mb-5">
          Your Reserved Airplanes
        </h1>

        {userAeroplanes && userAeroplanes.length > 0 ? (
          <ul className="list-unstyled d-flex flex-wrap">
            {userAeroplanes.map((aeroplane) => (
              <AeroplaneItem key={aeroplane.id} aeroplane={aeroplane} />
            ))}
          </ul>
        ) : (
          <p>No reserved aeroplanes found.</p>
        )}
      </div>
    </div>
  );
};

export default Reservations;
