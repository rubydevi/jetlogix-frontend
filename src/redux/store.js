import { configureStore } from '@reduxjs/toolkit';
import aeroplanesReducer from './aeroplanes/aeroplanesSlice';
import reservationsReducer from './reservations/reservationSlice';

const store = configureStore({
  reducer: {
    aeroplanes: aeroplanesReducer,
    reservation: reservationsReducer,
  },
});

export default store;
