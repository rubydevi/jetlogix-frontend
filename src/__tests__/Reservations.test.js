import React from 'react';
import { render, act } from '@testing-library/react';
import { BrowserRouter } from 'react-router-dom';
import { Provider } from 'react-redux';
import store from '../redux/store';

import Reservations from '../components/Reservations';

test('renders Reservations component correctly', async () => {
  await act(async () => {
    render(
      <Provider store={store}>
        <BrowserRouter>
          <Reservations />
        </BrowserRouter>
      </Provider>,
    );
  });
});
