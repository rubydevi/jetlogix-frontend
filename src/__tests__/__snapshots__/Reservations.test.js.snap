// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Reservations component correctly 1`] = `
<div>
  <div>
    <div
      class="col-12 text-center"
    >
      <h3
        class="fw-normal"
      >
         Reservations
      </h3>
    </div>
    <div
      class="d-flex justify-content-center align-items-center"
    >
      <p>
        No reserved aeroplanes found.
      </p>
    </div>
  </div>
</div>
`;
